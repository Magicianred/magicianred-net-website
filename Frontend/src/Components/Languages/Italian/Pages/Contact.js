import React, { Component } from 'react';
import axios from 'axios';
import Jumbotron from 'react-bootstrap/Jumbotron';

import Layout from './../Partials/Layout';

import Paragraph from '../../../../Models/Paragraph';

const publicUrl = process.env.PUBLIC_URL;

/**
 * Component for showing the Contact Page page.
 * 
 * @component
 * @example
 * return (
 *   <Contact />
 * )
 */
class Contact extends Component {
    
    constructor(props) {
        super(props);
        this.state = {
            paragraphs: [],
            hasError: false,
        };
    }        

    componentDidMount() {
        const paragraphs = [];
        this.serverRequest = 
            axios
                .get(`${publicUrl}/data/languages/english/pages/contact.json`)
                .then((result) => {
                    result.data[0].paragraphs.map((item, index) => paragraphs.push(new Paragraph(index, item.text)));
                    this.setState({ paragraphs });
            });
    }

    // componentWillUnmount() {
    //     if (this.serverRequest) {
    //         this.serverRequest.abort();
    //     }
    // }

    render() {
        const { items, paragraphs, hasError } = this.state;
        return (
            <Layout>
                {hasError ? <h1>Something went wrong.</h1> : null}
                <div className="page">
                    <Jumbotron className="page contact">
                        <h2>Contact Me page</h2>
                        <p>This is the Contact me page</p>
                        <hr />
                        <div>
                            {paragraphs.map(paragraph => paragraph.toDisplay())}
                        </div>
                    </Jumbotron>
                </div>
            </Layout>
        );
    }
}

export default Contact;