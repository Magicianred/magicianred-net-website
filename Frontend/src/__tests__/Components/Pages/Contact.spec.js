import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { create } from 'react-test-renderer';

import Contact from "./../../../Components/Pages/Contact";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

describe("Show the Contact Us page", () => {

    describe("Snapshots", () => {
    
        it("Matches the snapshot (with default title)", () => {
            const instance = create(<Contact />);
            expect(instance.toJSON()).toMatchSnapshot();
        });
    
    });

    it("Show about page", () => {
        act(() => {
            render(<Contact />, container);
        });
        expect(container.querySelector("h2").textContent).toBe("Contact page");
        expect(container.querySelector("p").textContent).toBe("This is the contact page");
    });

    xit("Show form module to send message", () => {

    });

    xit("Form module cannot send if miss require fields", () => {

    });
});