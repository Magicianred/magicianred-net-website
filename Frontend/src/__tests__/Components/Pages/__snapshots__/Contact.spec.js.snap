// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show the Contact Us page Snapshots Matches the snapshot (with default title) 1`] = `
<div
  className="page"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-12"
      >
        <h2>
          Contact page
        </h2>
        <p>
          This is the contact page
        </p>
        <hr />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <form
          className=""
          noValidate={true}
        >
          <div
            className="row"
          >
            <div
              className="col"
            >
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                  htmlFor="firstName"
                >
                  First Name
                </label>
                <input
                  className="form-control"
                  id="firstName"
                  onChange={[Function]}
                  placeholder="First name"
                  required={true}
                />
                <small
                  className="text-muted form-text"
                >
                  <span
                    className="badge badge-danger"
                  >
                    Field is required
                  </span>
                </small>
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                  htmlFor="lastName"
                >
                  Last Name
                </label>
                <input
                  className="form-control"
                  id="lastName"
                  onChange={[Function]}
                  placeholder="Last name"
                  required={true}
                />
                <small
                  className="text-muted form-text"
                >
                  <span
                    className="badge badge-danger"
                  >
                    Field is required
                  </span>
                </small>
              </div>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col"
            >
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                  htmlFor="email"
                >
                  Email address
                </label>
                <input
                  className="form-control"
                  id="email"
                  onChange={[Function]}
                  placeholder="Enter your email"
                  required={true}
                  type="email"
                />
                <small
                  className="text-muted form-text"
                >
                  <span
                    className="badge badge-danger"
                  >
                    Field is empty or invalid
                  </span>
                </small>
              </div>
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                  htmlFor="message"
                >
                  Say something to us
                </label>
                <textarea
                  className="form-control"
                  id="message"
                  onChange={[Function]}
                  required={true}
                  rows="3"
                />
                <small
                  className="text-muted form-text"
                >
                  <span
                    className="badge badge-danger"
                  >
                    Field is required
                  </span>
                </small>
              </div>
              <div
                className="form-group"
              >
                <div
                  className="form-check"
                >
                  <input
                    className="form-check-input"
                    disabled={false}
                    id="privacy"
                    onChange={[Function]}
                    required={true}
                    type="checkbox"
                  />
                  <label
                    className="form-check-label"
                    htmlFor="privacy"
                    title=""
                  >
                    Accept your privacy condition
                  </label>
                </div>
                <small
                  className="text-muted form-text"
                >
                  <span
                    className="badge badge-danger"
                  >
                    Field is required
                  </span>
                </small>
              </div>
              <button
                className="btn btn-primary"
                disabled={true}
                onClick={[Function]}
                type="button"
              >
                Send message
              </button>
              <br />
              <span
                className="badge badge-danger"
              >
                You have errors in form
              </span>
            </div>
          </div>
        </form>
      </div>
      <div
        className="col-sm-6"
      >
        <div
          className="contact jumbotron"
        >
          <div>
            <p>
              This form do not send any emails, this is only for a test purpose.
            </p>
            <h4>
              Show the data you would send
            </h4>
            <dl>
              <dt>
                First name
              </dt>
              <dd />
              <dt>
                Last name
              </dt>
              <dd />
              <dt>
                Email
              </dt>
              <dd />
              <dt>
                Message
              </dt>
              <dd>
                <p />
              </dd>
              <dt>
                Privacy
              </dt>
              <dd>
                no
              </dd>
            </dl>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
